src/services/transactions_details.rs

```diff
@@ -99,36 +98,56 @@ pub fn get_transactions_details(
     context: &Context,
     details_id: &String,
 ) -> ApiResult<TransactionDetails> {
+    let id_parts = parse_id(details_id)?;
+
+    match id_parts {
+        TransactionIdParts::Ethereum {
+            safe_address,
+            transaction_hash,
+            details_hash
+        } => get_ethereum_transaction_details(context, &safe_address, &transaction_hash, &details_hash),
+        TransactionIdParts::Module {
+            safe_address,
+            transaction_hash,
+            details_hash
+        } => get_module_transaction_details(context, &safe_address, &transaction_hash, &details_hash),
+        TransactionIdParts::Multisig { safe_tx_hash, .. } => get_multisig_transaction_details(context, &safe_tx_hash),
+        TransactionIdParts::TransactionHash(safe_tx_hash) => get_multisig_transaction_details(context, &safe_tx_hash),
+        _ => Err(ApiError::new_from_message(String::from("Bad transaction id"))),
```

Comment:
The only type missing in this match is creation, right? Should we be more explicit in the error? E.g. "Creation transactions are not supported for transaction details"

Reply:
Here this should probably be a `400` status code, as it is an invalud argument. Or a `422` as we could not process the id.